auth/libraries/pip-auth/authlib/validation.py

```diff
@@ -0,0 +1,26 @@
+import jwt
+
+from authlib.config import get_private_token_key, get_private_cookie_key
+
+
+def authenticate(token_encoded: str, cookie_encoded: str) -> int:
+    if token_encoded is None or cookie_encoded is None:
+        return 0
+    if not (is_bearer(token_encoded) and is_bearer(cookie_encoded)):
+        raise "Wrong format of token or cookie"
+    token_encoded = token_encoded[7:]
+    cookie_encoded = cookie_encoded[7:]
+    verify(cookie_encoded, get_private_cookie_key())
+    token = verify(token_encoded, get_private_token_key())
+    return token.get("userId", 0)
+
+
+def verify(encoded_jwt: str, secret: str) -> dict:
+    try:
+        return jwt.decode(encoded_jwt, secret)
+    except Exception as e:
+        raise f"JWT is wrong: {e}"
```

Comment:
DO not raise strings. Make custom exceptions and export them to the user of the lib to catch them. Maybe just the exceptions you defined in the docs in the main repo.